pulp_ansible/app/viewsets.py

```diff
@@ -1189,27 +1187,20 @@ class AnsibleDistributionViewSet(DistributionViewSet, RolesMixin):
     }
 
 
-class TagViewSet(NamedModelViewSet, mixins.ListModelMixin):
+class TagViewSet(viewsets.GenericViewSet, mixins.ListModelMixin):
     """
     ViewSet for Tag models.
     """
 
     endpoint_name = "pulp_ansible/tags"
-    queryset = Tag.objects.all()
+    queryset = (
+        CollectionVersion.objects.all()
+        .annotate(tag=Func(F("tags"), function="unnest"))
+        .values("tag")
+        .values(count=Count("pk"), name=F("tag"))
```

Comment:
I know I already approved this in the last PR, but sometimes SQL is straight magic...

Reply:
Well this is sql through the lenses of Django ORM.
The one thing that makes it really weird is the way "unnest" works.
It explodes the arrayfield into individual rows, a bit like left join.